src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -245,10 +245,27 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
                     var assembly = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
                     if (assembly != null)
                     {
-                        AddRange(allSymbolReferences, await finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact).ConfigureAwait(false));
+                        findTasks.Add(finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact));
                     }
                 }
+            }
+
+            while (findTasks.Count > 0)
+            {
+                // Keep on looping through the 'find' tasks, processing each when they finish.
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
So if cancellation gets requested between here and the next line, we won't observe the cancellation until at least one of the findTasks completes? Can we be more eager there?